packages/client/mutations/EndDraggingReflectionMutation.ts

```diff
@@ -87,6 +87,42 @@ const handleAddReflectionGroupToGroups = (store, reflectionGroup) => {
   addNodeToArray(reflectionGroup, meeting, 'reflectionGroups', 'sortOrder')
 }
 
+const handleAddReflectionToSpotlight = (
```

Comment:
Hey @mattkrick, I've got two architectural questions I'd like your advice on:

1. Should the remote reflection position begin where the remote user is dragging the reflection, i.e. the kanban, or in the Spotlight? 

Currently, it begins in the kanban (see gif). This tells the user it's being dragged by a teammate but maybe the position feels a bit random.

2. Should the results be groupable/ungroupable?
Enrique mentioned [here](https://github.com/ParabolInc/parabol/issues/4952#issuecomment-862513558) that we should prevent users from grouping/ungrouping the results.
This causes a problem in the following scenario:
- I have the Spotlight open
- another user ungroups a group on their browser that's in my Spotlight results
- that creates a new reflection group id which does not exist in the similarReflectionGroups query
- in my Spotlight, it looks like the card disappears

In this PR, if a results group is ungrouped, I'm adding the newly created reflection group id to the similarReflectionGroups query. This way, the card doesn't disappear and instead remains on the Spotlight.

Alternatively, if the remote reflection begins in the Spotlight, perhaps it could be dragged to the kanban and then disappear from the Spotlight. To do this, I'd try using the ref of the Spotlight card [here](https://github.com/ParabolInc/parabol/blob/0ef8108f6fc8cfce1c59d2fcd40d65a362acee4b/packages/client/components/ReflectionGroup/RemoteReflection.tsx#L63) instead of the `remoteDrag` position.

![remote-reflections](https://user-images.githubusercontent.com/39854876/133319375-d5ae997f-d866-42df-b334-72945fb6d886.gif)

Note: I'll create a separate PR with a better Spotlight groups UI

Reply:
I think the answers to both 1 & 2 are yes. I've pushed an update following the remote reflection plan I outlined here. In the Spotlight UI groups PR, I'll work on adding new reflection groups to the Spotlight column with the least groups

Reply:
> Should the remote reflection position begin where the remote user is dragging the reflection, i.e. the kanban, or in the Spotlight?

The trick here is a single source of truth, ie only 1 card per reflection.
The logic I'd use is something like this:

- when the viewer has spotlight open & a remote user starts to drag a card, make sure the remote user is not dragging the target reflection. if they are & the other user wins the arbitration, close the spotlight. see https://github.com/ParabolInc/parabol/blob/8c68bf771a350b31c7948e6bd11c5c110e30c977/packages/client/mutations/StartDraggingReflectionMutation.ts#L71-L82
- when the remote user is dragging a card, check to see if that card exists inside the spotlight. 
- if it does, animate the card in front of the spotlight modal
- it it does not, check the `remoteDrag.targetId` 
- if the targetId is null or the targetId does not exist in the spotlight, animate the remote reflection behind the spotlight modal
- if the targetId exists in the spotlight, animate the remote reflection relative to the reflection.

In plain english, let's assume you have a spotlight modal open. if a remote user is dragging a card `A` & hovering over `B`, then you should see that I'm hovering over `B` if it's in your spotlight results. that way when the remote user drops it, it doesn't seem like magic just happened.  

> Should the results be groupable/ungroupable?

- yes, you can drag a result card onto the target card OR you can drag the target card onto a result card.
- like you say, if a remote user drags a result card `A` onto a card that doesn't exist in spotlight `B`, then the spotlight should remove group `A` from the results & add group `B` in its place. visually, it'll look like the card just grew a stack behind it.
- if a remote user drags a card that doesn't exist in spotlight `B` onto a card that _does_ exist in spotlight results `A`, then the user should see that card hovering & then eventually dropped onto `A` (the logic is the same as your question above)